extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/DefaultTokenStateManager.java

```diff
@@ -149,7 +159,15 @@ public Uni<AuthorizationCodeTokens> getTokens(RoutingContext routingContext, Oid
                     try {
                         accessTokenExpiresIn = accessTokenData[1].isEmpty() ? null : Long.valueOf(accessTokenData[1]);
                     } catch (ArrayIndexOutOfBoundsException ex) {
-                        return Uni.createFrom().failure(new AuthenticationCompletionException("Session cookie is malformed"));
+                        final String error = "Session cookie is malformed";
+                        LOG.debug(ex);
+                        // Make this error message visible in the dev mode
+                        return Uni.createFrom().failure(new AuthenticationFailedException(error));
+                    } catch (NumberFormatException ex) {
+                        final String error = "Access token expires_in property in the session cookie must be a number";
```

Comment:
It would be nice if we had the value that was found, for debugging purposes
```suggestion
                        final String error = "Access token expires_in property in the session cookie must be a number, found %s";
```

Reply:
@gastaldi Sure, took me a little while to experiment with getting to that value from different branches, so I wrapped the actual parsing cal in a function to make it easier, hope it looks better